jep/7/README.adoc

```diff
@@ -67,22 +69,68 @@ endif::[]
+This includes but not limited to:
 
-Workarounds in Jenkins core implemented specifically to enable `ruby-runtime` will be reviewed and considered for reversal.
-No future core development will consider the impact on `ruby-runtime` and plugins based on it, and bug reports to core and related components about their impact on `ruby-runtime` and plugins based on it will be closed as *Won't Fix*.
+* link:https://github.com/stapler/stapler/tree/master/jruby[Stapler JRuby library] bundled in the Jenkins core
+* Permit lists for link:/jep/200[JEP-200: Switch Remoting/XStream denylist to a permitlist]
+* XStream converters for JRuby components
 
+No future core development will consider the impact on Ruby/Python runtimes and plugins based on them.
+Bug reports to core and related components about their impact on `ruby-runtime` and plugins based on it will be closed as *Won't Fix*.
+
+=== Rollout plan
+
+At least 1 month before removal::
+
+* Announcement blogpost on jenkins.io with the user-focused overview of the change and the delcivery plan
+* Plugins are marked for adoption
+* Plugins will be deprecated in the Update site immediately after the announcement.
+  The deprecation notice will reference the announcement blogpost as justification.
+* Plugin users will have 1 month to bring up their concerns, if any.
+  They will have an option to step up and contribute to the plugins so that they are migrated to the standard Jenkins plugin runtime.
+
+Removal date::
+
+* Plugins are removed from official Jenkins update centers.
+  Users of both weekly and LTS release baselines will no longer be able to install or update the plugins.
+* Maybe: Additional announcement is published on jenkins.io to advice deletion of the plugins
+* If there are any unbpublished security issues related to the depublished plugins,
+  a security advisory is released for them shortly after depublishing.
+* Ruby and Python plugin runtime codebases are archived on GitHub.
+
+After the removal::
+
+A pull request is submitted to the Jenkins, with a cleanup of the code and functionality slated for deletion.
+Stapler JRuby library might be also deleted from the Stapler codebase.
+This is a breaking change.
+Once the pull request is merged, the next weekly release will no longer be able to run the Ruby/Python runtime based plugins properly,
+with undefined behavior and likely Jenkins startup failure as a result.
+This needs to be explicitly referenced in the changelog.
+
+Jenkins LTS users will be able to continue using the deprecated/depublished plugins until a new LTS baseline picks up the change.
+ETA for it - the September 2021 LTS release.
+After this release new Jenkins LTS versions will not be able to run the plugins as well.
+This change will be documented in the LTS upgrade guidelines.
+
+Plugins may be restored in the update centers once
+there are contributors who take ownership of a plugin and migrate it to the new runtime.
```

Comment:
"the new runtime"  what is the "new" runtime here?  is this just migrate it to the standard Java runtime?

Reply:
Yes, standard Java one. Will reword

Reply:
```suggestion
there are contributors who take ownership of a plugin and migrate it to the standard Java plugin runtime.
```